zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -521,11 +525,16 @@ private void receiveMessage(ByteBuf message) {
                             throw new IOException("Len error " + len);
                         }
                         ZooKeeperServer zks = this.zkServer;
-                        if (zks == null || !zks.isRunning()) {
-                            throw new IOException("ZK down");
+
+                        if (zks != null && !zks.isRunning()) {
```

Comment:
Is the condition correct? I'd assume that the condition would be `zks.isRunning()`.
```suggestion
                        if (zks != null && zks.isRunning()) {
```

Reply:
Thanks, I was not sure if this was needed and changed the condition.